### ISSUE TITLE
Revoked privileges cause problems with privilege-dependent items

### ISSUE THREAD

**Issue by SwissalpS** [mt-mods/otp#1]
see: https://github.com/pandorabox-io/in-game/issues/324

in a nutshell: some items are priv dependant and get removed during login as privs are revoked.

**Comment by S-S-X** [comment]
To reduce significantly what can be done while still having privileges could be:
1. revoke just interact, keep all other privileges.
2. override chat command handler to do simple player name check, if player is in middle of otp authentication just cancel all commands (or possibly just allow few whitelisted commands).

That should keep it compatible with almost everything as it does not need mod / command / thing specific overrides and will keep privileges but instead directly removes actual functionality until auth is done.

Freezing player / protecting player in addition to this will be useful too. One bit complicated but fool proof mechanism would be single safe protected location in map where players would be teleported on login and teleported back to original position immediately after auth is done.
Complicated because it involves actual protected in world location, kind of like what those jail mods offer.

**Comment by BuckarooBanzay** [comment]
> revoke just interact, keep all other privileges.

~~you could still execute malicious chat-commands that way :/ (`//lua` comes to mind)~~

> in a nutshell: some items are priv dependant and get removed during login as privs are revoked.

~~my proposal (because the moderator-armor does the drop-action) would be to add an api to the `otp` mod and let
the `moderator_armor` fetch the revoked privs of the player if they are in an otp session~~

**Comment by S-S-X** [comment]
> you could still execute malicious chat-commands that way :/ (//lua comes to mind)

You probably missed my proposal to override all chat command handlers and disable every chat command directly?

So you could run any chat command but every command would return say _Chat commands are disabled until authentication is completed_. Not 100% sure about this but It should be as simple as swapping first `core.registered_on_chat_messages` to handler that checks if player has entered otp or not and cancels real first handler returning error message.

I'm pretty sure it is single entry point for every possible chat input unless if there is some very special commands implemented in engine c++ core, not aware of any such thing.

**Comment by BuckarooBanzay** [comment]
> You probably missed my proposal to override all chat command handlers and disable every chat command directly?

yeah, i was in a hurry and missed that part, sorry. This does make sense :+1:
